### PR TITLE
add ut for cache

### PR DESCRIPTION
**Pull request by allenhaozi**
Signed-off-by: mahao <allenhaozi@gmail.com>

### Ⅰ. Describe what this PR does

ut for cache and sync dependency in vendor

### CODE REVIEW

**Comment by allenhaozi** [comment]
for a simple mock Alluxio master reports
import the package GoConvey
Please assess the benefits of introducing this module and whether we have such a test method

@cheyang @TrafalgarZZZ @littletiger123 @xieydd @yangyuliufeng 

thank you

**Comment by cheyang** [comment]
> for a simple mock Alluxio master reports
> import the package GoConvey
> Please assess the benefits of introducing this module and whether we have such a test method
> 
> @cheyang @TrafalgarZZZ @littletiger123 @xieydd @yangyuliufeng
> 
> thank you

I think it's helpful for ut for we can support some complex scenarios. Thanks.

File: pkg/ddc/alluxio/cache.go [hunk 1]
```diff
@@ -17,94 +17,74 @@ package alluxio
...
+
+var (
+	cachedPercentageFormat = "%.1f%%"
 )
 
 // queryCacheStatus checks the cache status
 func (e *AlluxioEngine) queryCacheStatus() (states cacheStates, err error) {
-	summary, err := e.reportSummary()
+	// get alluxio fsadmin report summary
+	summary, err := e.GetReportSummary()
```
Comment (cheyang) [inline pkg/ddc/alluxio/cache.go:35]: May I know why do you change `reportSummary`  from private to public?

File: pkg/ddc/alluxio/cache.go [hunk 2]
```diff
@@ -17,94 +17,74 @@ package alluxio
...
-	// `dataset.Status.UfsTotal` probably haven't summed, in which case we won't compute cache percentage
-	if dataset.Status.UfsTotal != "" && dataset.Status.UfsTotal != METADATA_SYNC_NOT_DONE_MSG {
-		usedInBytes, _ := utils.FromHumanSize(states.cached)
-		ufsTotalInBytes, _ := utils.FromHumanSize(dataset.Status.UfsTotal)
-		states.cachedPercentage = fmt.Sprintf("%.1f%%", float64(usedInBytes)/float64(ufsTotalInBytes)*100.0)
-	}
+	e.patchDatasetStatus(dataset, &states)
 
-	states.cacheHitStates = e.getCacheHitStates()
+	states.cacheHitStates = e.GetCacheHitStates()
```
Comment (cheyang) [inline pkg/ddc/alluxio/cache.go:56]: Same question.

File: pkg/ddc/alluxio/cache.go (hunk 1 above)
Comment (allenhaozi) [reply pkg/ddc/alluxio/cache.go:35]: because `Convey ` can not mock private function

File: pkg/ddc/alluxio/cache.go (hunk 2 above)
Comment (allenhaozi) [reply pkg/ddc/alluxio/cache.go:56]: because the current version of `Convey` cannot mock private functions

File: pkg/ddc/alluxio/cache.go (hunk 1 above)
Comment (cheyang) [reply pkg/ddc/alluxio/cache.go:35]: I see. Thanks for explanation.

**Pull request merged**